src/Sensors.cpp

```diff
@@ -895,6 +897,22 @@ void Sensors::PMGCJA5Init() {
 void Sensors::dhtInit() {
 }
 
+// Altitude compensation for CO2 sensors without Pressure atm or Altitude compensation
+
+void Sensors::CO2correctionAlt() {
+    DEBUG("-->[SENSORS] Altitud Compensation for CO2 lectures ON");
+    DEBUG("-->[SENSORS] CO2 original: ", String(CO2).c_str());
+    alt = getCO2AltitudeCompensation();
+    float hpa = 1012 - 0.118 * alt + 0.00000473 * alt * alt;            // Cuadratic regresion formula obtained PA (hpa) from high above the sea
+    DEBUG("-->[SENSORS] Atmospheric pressure calculated in hPa: ", String(hpa).c_str());
+    float CO2cor = (0.016 * ((1013.25 - hpa) /10 ) * (CO2 - 400)) + CO2;       // Increment of 1.6% for every hpa of difference at sea level
+    CO2 = round (CO2cor);
+}
+
+float Sensors::getCO2AltitudeCompensation(){
+    return alt;
+}
```

Comment:
this method it is the same than current `getAltitude()` method, it doing the same. I think that maybe only remove this method, and in the CO2correctionAlt() use a new global variable, for example `alt_comp` variable, and also write a new method `setCO2AltitudeCompensation(float altitude)` where you set `alt_comp`. With these changes will be have `alt` for real altitude, for example the current bme680, and alt_comp for sensors that need it.

Reply:
Done